packages/core/src/model.js

```diff
@@ -2282,16 +2286,14 @@ ${associationOwner._getAssociationDebugList()}`);
                 continue;
               }
 
-              if (Object.prototype.hasOwnProperty.call(result, key)) {
-                const record = result[key];
-
-                const attr = find(
-                  modelDefinition.attributes.values(),
-                  attribute => attribute.attributeName === key || attribute.columnName === key,
-                );
-
-                instance.dataValues[attr && attr.attributeName || key] = record;
-              }
+              const value = result[key];
+              const attr = find(
+                modelDefinition.attributes.values(),
+                attribute => attribute.attributeName === key || attribute.columnName === key,
+              );
```

Comment:
This changed was needed because the primary key was returned as a bigint and was used as-is instead of being parsed by the data type.